### ISSUE TITLE
Ingress-nginx make SPA 404 when refresh browser

### ISSUE THREAD

**Issue by KarateJB** [KarateJB/JB-eBooks#75]
### Environment:

Helm v3 Chart that serves Angular application with ingress-nginx

### Problem

It's fine to browse the website by router, but results in 404 by refreshing the browser. The issue doesn't happen when running it by docker compose.

**Comment by KarateJB** [comment]
Here are some references

- [Angular application Deployed on Kubernetes with Nginx Ingress refresh problem](https://angularquestions.com/2021/05/29/angular-application-deployed-on-kubernetes-with-nginx-ingress-refresh-problem/)
- [angular app in kubernetes using nginx ingress path](https://stackoverflow.com/questions/62009062/angular-app-in-kubernetes-using-nginx-ingress-path)
- [Angular on Kubernetes (ingress routing)](https://stackoverflow.com/questions/56989555/angular-on-kubernetes-ingress-routing)
- [Vuejs into kubernetes cluster and nginx ingress returns 404 when refreshed](https://stackoverflow.com/questions/61920045/vuejs-into-kubernetes-cluster-and-nginx-ingress-returns-404-when-refreshed)
- [Rewrite annotations](https://github.com/kubernetes/ingress-nginx/tree/main/docs/examples/rewrite)

After trying the above possible solutions, I found that

The following manifest will still make refreshing a non-root URL 404. 

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: demo-k8s-web
  namespace: demo-k8s
  annotations:
    nginx.ingress.kubernetes.io/use-regex: "true"
    nginx.ingress.kubernetes.io/rewrite-target: /$1
spec:
  rules:
    - host: "demo-k8s-web.com"
      http:
        paths:
          - path: /(.*)
            pathType: Prefix
            backend:
              service:
                name: demo-k8s-web
                port:
                  number: 80
```

The following manifest will not make refreshing a non-root URL 404, but the component cannot be correctly loaded.

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: demo-k8s-web
  namespace: demo-k8s
  annotations:
    nginx.ingress.kubernetes.io/rewrite-target: /
spec:
  rules:
    - host: "demo-k8s-web.com"
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: demo-k8s-web
                port:
                  number: 80
```

The solution that can solve the 404 issue (so far I know) is using the `HashLocationStrategy` in `app-routing.module.ts`.

```ts
@NgModule({
  imports: [RouterModule.forRoot(routes, {useHash: true})],
  exports: [RouterModule]
})
```

**Comment by KarateJB** [comment]
Closed by #77
